### PR TITLE
Use node 13 for CodeGeneration and merge pipelines

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: eng/Directory.Build.Data.targets
Comment (pakrym) [inline eng/Directory.Build.Data.targets:32]: Don't print every changelog entry.

File: eng/scripts/Export-API.ps1
Comment (pakrym) [inline eng/scripts/Export-API.ps1:10]: Faster API export

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [reply eng/Directory.Build.Data.targets:32]: While this is kind of annoying for the core build for individual builds the idea is that people will see the change log entry when they build. At any rate I'm fine with this as well.

Comment (pakrym) [reply eng/Directory.Build.Data.targets:32]: I'm not sure if they ever look in the build step for this.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (weshaggard) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:52]: I'm fine with you splitting out the code generation and making it run in parallel but I'd like to keep the other stuff in the Analyze step for consistency across languages and to provide a place to run static analysis type validation. Currently it doesn't look like it is even downloading the packages but it should be for the Component Governance check. @chidozieononiwu can you please take a look at that?

Comment (pakrym) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:52]: Do you want me to revert it?

Comment (pakrym) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:52]: > run static analysis type validation

In addition, we run most of that (roslyn analyzers) during build

Comment (weshaggard) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:52]: Yes please revert back to keeping the Analyze job, I know there are at least some things like Component Governance reporting that only looks at our Analyze jobs for reporting purposes. 

As for static analyzers I'm not necessarily talking about the code analyzers that would be run as part of compiling. I'm more meaning other policies like readme/link checks/component governance/ (possibly policheck and credscan in the future).

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [reply eng/Directory.Build.Data.targets:32]: I agree, most folks don't probably look at it.

**Review by weshaggard — APPROVED** [review]
Looks good but lets please wait until we finish up the releases this week before merging.

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (mitchdenny) [inline eng/pipelines/templates/jobs/archetype-sdk-client.yml:83]: Is there a reason why this can't be run within the Analyze job (on Windows)? Adding this job here causes a whole agent to be consumed.

**Review by mitchdenny — DISMISSED** [review]

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (pakrym) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:83]: Analyze runs after build completes and we would like to start the code check as early as possible because it's the longest leg of the build. In my initial change, I removed the Analyze job and moved all the steps to build to reduce the agent found but @weshaggard wanted to preserve it.

Comment (weshaggard) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:83]: @mitchdenny This is a pretty long running task which would be good to run in parallel with the build and test legs. I believe the goal was to have the Analyze job run after the build job which makes the overall pipeline run take longer then it should. The original reason was because we wanted to download and analyze the packages from the build job, however it looks like we aren't even downloading the packages any more and as such the [Component Governance](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=536064&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=72a3110e-44c0-5323-690d-8dd09cc84ac8) is not running on any packages beyond what is committed in the repo (yes there are still a couple of those :(). I just looked at our [CG runs](https://dev.azure.com/azure-sdk/internal/_componentGovernance/98522?_a=components&typeId=1937810&alerts-view-option=active) and the packages are present and it looks like they are getting analyzed in our [dev publishing step](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=536064&view=logs&j=b1e79959-24d8-5aa9-2799-72d40c3e051b&t=f3fc8d9f-4e68-567c-1c63-cd6d9a749ff5) instead.  

Given that @pakrym can you just keep this in the Analyze job and just remove the Depends on Build from that job. 

I've filed https://github.com/Azure/azure-sdk-for-net/issues/15143 to follow-up on the CG stuff.

Comment (pakrym) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml:83]: >Given that @pakrym can you just keep this in the Analyze job and just remove the Depends on Build from that job.

CodeGeneration is significantly faster on ubuntu, that's part of the speedup.

**Review by weshaggard — APPROVED** [review]

File: eng/pipelines/templates/jobs/archetype-sdk-client.yml
Comment (weshaggard) [reply eng/pipelines/templates/jobs/archetype-sdk-client.yml]: I'm fine with the move to ubuntu as well.

**Pull request merged**
